Core/contentblocker.js

```diff
@@ -114,12 +119,22 @@ var duckduckgoContentBlocking = function() {
 	}
 
 	// private
-	function isFirstParty(event) {
+	function isAssociatedFirstPartyDomain(event) {
 		var topLevelUrl = getTopLevelURL()	
-		var url = toURL(event.url, topLevelUrl.protocol)
-		if (url != null && domainsMatch(url, topLevelUrl)) {
+
+		var urlToCheck = toURL(event.url, topLevelUrl.protocol)
+		if (urlToCheck == null) {
+			return false
+		}
+
+		if (domainsMatch(urlToCheck, topLevelUrl)) {
+			return true
+		}
+
+		var result = DisconnectMe.parentTracker(urlToCheck)
+		if (result && domainsMatch(result.parent, topLevelUrl)) {
```

Comment:
This is comparison throws and error and never completes

Reply:
Do the extension do this in both directions? e.g
• instagram is a child of facebook, so allow an instagram tracker on facebook
• instagram is a child of facebook, so allow a facebook tracker on instragram

We currently cover the top case not the second, just want to make sure we're consistent.

Reply:
Thanks.  I've updated the logic to make sure we cover both cases and improved error handling for security errors thrown in iframes.   We should probably run the report after this is merged ... :)